src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        public TextLoaderRange(int ordinal)
+        {
+            Min = ordinal;
+            Max = ordinal;
+        }
+
+        public TextLoaderRange(int min, int max)
+        {
+            Min = min;
```

Comment:
Please refer to \src\microsoft.ml.data\dataloadsave\text\textloader.cs line 149 - isValid method

---
In reply to: [188737311](https://github.com/dotnet/machinelearning/pull/142#discussion_r188737311) [](ancestors = 188737311)

Reply:
@codemzs It's https://github.com/dotnet/machinelearning/blob/c023727b76970ab913ec1ce38276508835c17bcf/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L149 I suppose. While there is this function, it would make sense to check invariants "on the spot" _also_ (if these cases really are not allowed). Now it's not possible to see this invariant here, locally, just by looking at this code and correspondginly have an error "on the spot" for erroneous case. It also creates a precedent to write code that is more brittle in that if something is changed "somebody else somewhere checks for the validity". It's very difficult to anyone outside of the project, or new to project, make any changes without reading a lot of code (which likely means it won't happen).

Reply:
I have added the checks for invariants here but I'm not sure if this is really needed because the checks are already performed by the same class since it has a partial implementation that calls into isValid method as I have previously pointed. 

---
In reply to: [188838661](https://github.com/dotnet/machinelearning/pull/142#discussion_r188838661) [](ancestors = 188838661)